### PR TITLE
Rename Component to Schema

### PR DESCRIPTION
**Pull request by kellpossible**
As outlined in the discussion https://github.com/juhaku/utoipa/pull/215#discussion_r921380613

Public facing breaking changes:
+ Rename `Component` enum to `Schema`.
+ Rename `Component` trait to `ToSchema`.
+ Rename `ComponentType` to `SchemaType`.
+ Rename `ComponentFormat` to `SchemaFormat`.
+ Rename `#[component(...)]` struct/enum/field attributes to `#[schema(...)]`.
+ Rename `Property` field `component_type` to `schema_type`.
+ Rename `Object` field `component_type` to `schema_type`.
+ Rename `Array` field `component_type` to `schema_type`.
+ Rename `Ref` method `from_component_name()` to `from_schema_name()`.
+ `ComponentBuilder`'s methods `schema()` and `schema_from_iter()` now accept `Into<RefOr<Schema>>` instead of `Into<Schema>`, so schema components can either be references or inline definitions. For calls which were passing `Into<Schema>` implementations like `ObjectBuilder` (e.g. `.schema(ObjectBuilder::new())`), now needs to be wrapped in `.schema(Schema::from(ObjectBuilder::new()))` in order to resolve the type.
+  Change `#[openapi(components(...), responses(...)]` to `#[openapi(components(schemas(...), responses(...))]`.

Internal changes:
+ Rename `ComponentAttr` to `SchemaAttr`.

TODO:

 - [x] Rename the `utoipa-gen/src/schema/component` module/s
 - [x] Implement proposed restructure of the `openapi` derive attribute https://github.com/juhaku/utoipa/pull/215#discussion_r929280545
 - [x] Make `Components` accept `RefOr` for `schemas` field, so schema components can be references.

### CODE REVIEW

**Comment by kellpossible** [comment]
@juhaku how does this look? It's not completely ready, but before I put the work in to polish it up and fix any broken compilation and tests, wanted to get your feedback.

File: tests/component_derive_test.rs [hunk 1]
```diff
@@ -537,7 +537,7 @@ fn derive_with_box_and_refcell() {
 
 #[test]
 fn derive_struct_with_inline() {
-    #[derive(utoipa::Component)]
+    #[derive(utoipa::derive_to_schema)]
```
Comment (juhaku) [inline tests/component_derive_test.rs:540]: Shouldn't this be `utoipa::ToSchema` ?

**Review by juhaku — COMMENTED** [review]
Wow.. this is crude work. Looks pretty good. There was couple of cases where `derive_to_schema` should have been `utoipa::ToSchema`.

**Review by juhaku — APPROVED** [review]
:trophy: Super

**Comment by kellpossible** [comment]
@juhaku thanks! Still a couple more TODO items and this will be ready, hopefully in about 30 minutes :slightly_smiling_face:

**Review by juhaku — COMMENTED** [review]
:+1:

**Comment by juhaku** [comment]
Rename in `scripts/test.sh` `component_derive_no_serde_json` to the actual name of the test. I suppose it is now `schema_derive_no_serde_json` same for CI in `.github/workflows/build.yaml` :)

**Comment by kellpossible** [comment]
> Rename in scripts/test.sh component_derive_no_serde_json to the actual name of the test. I suppose it is now schema_derive_no_serde_json same for CI in .github/workflows/build.yaml :)

Fixed!

**Review by juhaku — APPROVED** [review]
Great.. hard work of you :slightly_smiling_face:  Many thanks :heart:

**Pull request merged**